platform/ios/test/MGLMapViewLayoutTests.m

```diff
@@ -67,7 +67,7 @@ - (void)testOrnamentPlacement {
     CGFloat bottomSafeAreaInset = 0.0;
     double accuracy = 0.01;
 
-    if ( [self.mapView respondsToSelector:@selector(safeAreaInsets)] ) {
+    if (@available(iOS 11.0, *)) {
```

Comment:
As an ObjC old-timer, changes like this always give me pause. I guess it's just the newer way of doing things, but I find the old explicit `respondsToSelector:` API availability check more comprehensible than the "what version are we running on" check. Informal protocols are a waning concept I guess...

Nothing to change here per se, just sharing my thoughts...

Reply:
Aye, this is definitely a bit of a shift in philosophy — I swapped to `@available()` here because `-Wunguarded-availability-new` will complain about just doing a selector-based check. We could certainly do both (and that’s technically safer, of course), but it feels redundant to me for system API.

Reply:
> `-Wunguarded-availability-new` will complain about just doing a selector-based check

Interesting. Sounds like a slow forced transition is underway, and we do need to nudge the codebase forward toward conformance around the newer pattern.

I agree with you re: redundancy, especially in tests... we should continue to keep an eye out for the right balance of style conformance and readability/comprehensibility, especially in the implementation.